python/ray/data/impl/simple_block.py

```diff
@@ -71,7 +71,7 @@ def random_shuffle(self, random_seed: Optional[int]) -> List[T]:
 
     def to_pandas(self) -> "pandas.DataFrame":
         import pandas
-        return pandas.DataFrame(self._items)
+        return pandas.DataFrame(self._items).rename(columns=str)
```

Comment:
Comment on why the rename?

Reply:
+1

Reply:
If there's no rename, `test_map_batch` will fail at https://github.com/ray-project/ray/blob/4f9aceb3a6ed3f374b5ecab3157cc5dad742ea97/python/ray/data/tests/test_dataset.py#L2046-L2051 with error message

```
E                       ray.exceptions.RayTaskError(ValueError): [36mray::_map_block_nosplit()[39m (pid=291702, ip=11.159.247.24)
E                         File "/home/admin/dev/github/alipay/ray/python/ray/data/impl/compute.py", line 51, in _map_block_nosplit
E                           input_files=input_files, exec_stats=stats.build())
E                         File "/home/admin/dev/github/alipay/ray/python/ray/data/block.py", line 184, in get_metadata
E                           schema=self.schema(),
E                         File "/home/admin/dev/github/alipay/ray/python/ray/data/impl/pandas_block.py", line 95, in schema
E                           "A Pandas DataFrame with column names of non-str types"
E                       ValueError: A Pandas DataFrame with column names of non-str types is not supported by Ray Dataset. Column names of this DataFrame: [0].
```

Since we disallow non-str column names for Ray datasets, we must convert non-str type column names to str type column names when converting from a simple block dataset to a pandas block dataset. So I added the column name conversion in `SimpleBlockAccessor.to_pandas`.

And due to this column name type enforcement, I have to modify test case `test_batch_tensors` and `test_to_arrow_refs`.

Do you have any better ideas? I'm aware that this may break something, but I can't think of a better way.

Reply:
@ericl What if we applied the `ray.data.range_arrow()` and `ray.data.range_tensor()` semantics here, where a single column under the column name `"value"` is created? I.e.
```suggestion
        return pandas.DataFrame({"value": self._items})
```

I think that this improves the consistency of how we do the list --> single-column table conversion, and then we don't have to worry about this issue.

Reply:
Yeah I'm in favor of this. I think it improves consistency of the API (0 vs value).

Reply:
Changed to "value".